### ISSUE TITLE
Problem transitioning from agent_pool_profile to default_node_pool

### ISSUE THREAD

**Issue by mark-greene** [terraform-providers/terraform-provider-azurerm#5160]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.17
provider "azurerm" (hashicorp/azurerm) 1.38.0

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

```hcl
# Original agent_pool_profile
  agent_pool_profile {
    name            = "default"
    count           = local.agent_count
    vm_size         = "Standard_DS2_v2"
    os_type         = "Linux"
    os_disk_size_gb = 100
    vnet_subnet_id  = azurerm_subnet.aks.id
  }

# New default_node_pool
  default_node_pool {
     availability_zones    = []
     enable_auto_scaling   = false
     enable_node_public_ip = false
     max_count             = 0
     max_pods              = 30
     min_count             = 0
     name                  = "default"
     node_count            = local.agent_count
     node_taints           = []
     os_disk_size_gb       = 100
     type                  = "AvailabilitySet"
     vm_size               = "Standard_DS2_v2"
     vnet_subnet_id        = azurerm_subnet.aks.id
  }
```

### Expected Behavior

I'm stuck with conflicting error message saying the value must be in the range of 1 - 100 and the value must be 0.  I expect one of the values to work.

### Actual Behavior

Error: expected default_node_pool.0.min_count to be in the range (1 - 100), got 0  
on main.tf line 139, in resource "azurerm_kubernetes_cluster" "aks":  
139: resource "azurerm_kubernetes_cluster" "aks" {  

Error: expected default_node_pool.0.max_count to be in the range (1 - 100), got 0  
on main.tf line 139, in resource "azurerm_kubernetes_cluster" "aks":  
139: resource "azurerm_kubernetes_cluster" "aks" {  

**When I set min_count and max_count to 1 I get**

Error: Error expanding `default_node_pool`: `max_count` and `min_count` must be set to `0` when enable_auto_scaling is set to `false`  
on main.tf line 139, in resource "azurerm_kubernetes_cluster" "aks":  
139: resource "azurerm_kubernetes_cluster" "aks" 

### Steps to Reproduce

1. `terraform apply`

**Comment by tombuildsstuff** [comment]
hi @mark-greene 

Thanks for opening this issue.

Taking a look at this since these fields are optional you should be able to omit these fields/set them to `null` which should have the same effect here - would you be able to take a look and see if that works for you? If so we can look to update the error message here to say "should be unspecified" rather than "should be 0"

Thanks!

**Comment by jibsheet** [comment]
We've run into the same issue because we allow users to specify `enable_auto_scaling = "${var.cluster_pool_enable_auto_scaling}"` as well as the max/min nodes.  Is there a reason not to allow 0?  The API requires 0.

**Comment by joebeadle** [comment]
Any update on this issue?

**Comment by PirateBread** [comment]
Also running into this issue. We have disabled autoscale which sets min/max count to zero.

Setting it to NULL:
```
var.aks_cluster.scale_min_count is "null"
Inappropriate value for attribute "min_count": a number is required.

```

**Comment by tombuildsstuff** [comment]
Fixed in v2.5 - `min_count` and `max_count` should now be set to `null` rather than `0`
